src/Workspaces/Core/Portable/Workspace/Solution/DocumentState.cs

```diff
@@ -393,42 +393,10 @@ public new DocumentState UpdateText(SourceText newText, PreservationMode mode)
                 throw new ArgumentNullException(nameof(newText));
             }
 
-            // check to see if this docstate has already been branched before with the same text.
-            // this helps reduce duplicate parsing when typing.
-            if (mode == PreservationMode.PreserveIdentity)
-            {
-                var br = _firstBranch;
```

Comment:
do we still need the _firstBranch field anymore?

Reply:
Oh, n/m, i see it was removed below.  bleagh for code that intermixes fields and nested types :D